### PR TITLE
Addon-actions: Add Storybook Args support

### PR DESCRIPTION
**Pull request by shilman**
Issue: N/A

## What I did

Storybook Args is a new format for stories that passes a dynamic data object "Args" into story functions. This can be used to automatically provide actions event callbacks into stories and components.

This PR adds two forms of Args compatibility using story parameters.
1) `argTypes.action` that adds an action for each matching argType
2) `actions.argTypesRegex` that adds an action for each arg type name that matches the given regex.

See the updated stories for examples.

## How to test

See attached tests and `official-storybook` stories

### CODE REVIEW

File: addons/actions/README.md [hunk 1]
```diff
@@ -18,10 +18,43 @@ Then, add following content to `.storybook/main.js`
...
+
+export default {
+  title: 'Button',
+  parameters: { actions: { args: ['onClick'] } },
+};
+
+export const defaultView = ({ onClick }) => <Button onClick={onClick}>Hello World!</Button>;
 ```
 
+Alternatively, suppose you have a naming convention, like `onX` for event handlers. The following configuration automatically creates actions for each `onX` argType (which you can either specify manually or generate automatically using [Storybook Docs](https://www.npmjs.com/package/@storybook/addon-docs).
```
Comment (tmeasday) [inline addons/actions/README.md]: Maybe add an example of how you'd provide an argType to the story? Or is that sort of pointless?

File: addons/actions/src/preset/addArgs.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+//   argTypesRegex?: RegExp;
+// }
+
+/**
+ * Automatically add action args for argTypes whose name
+ * matches a regex, such as `^on.*` for react-style `onClick` etc.
+ */
+export const inferActionsFromArgTypes: ParameterEnhancer = context => {
+  const { args, actions, argTypes } = context.parameters;
+  if (!actions || actions.disable || !actions.argTypesRegex || !argTypes) {
```
Comment (tmeasday) [inline addons/actions/src/preset/addArgs.ts]: Does this mean you need to ensure that this enhancer runs after docs'? That seems like a bit of a gotcha. We should document it at least

File: addons/actions/src/preset/addArgs.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  const argTypesRegex = new RegExp(actions.argTypesRegex);
+  const actionArgs = Object.keys(argTypes).reduce((acc, name) => {
+    if (argTypesRegex.test(name)) {
+      acc[name] = action(name);
+    }
+    return acc;
+  }, {} as Args);
+
+  return {
+    args: combineParameters(actionArgs, args),
```
Comment (tmeasday) [inline addons/actions/src/preset/addArgs.ts]: Why `combineParameters`? an action doesn't have any subfields, does it?

**Review by tmeasday — CHANGES_REQUESTED** [review]
Looks good, couple of little things

File: addons/actions/src/preset/addArgs.ts (hunk 3 above)
Comment (shilman) [reply addons/actions/src/preset/addArgs.ts]: Why not? Is it sufficiently slower that I should care?

File: addons/actions/src/preset/addArgs.ts (hunk 2 above)
Comment (shilman) [reply addons/actions/src/preset/addArgs.ts]: Updated docs

File: addons/actions/README.md (hunk 1 above)
Comment (shilman) [reply addons/actions/README.md]: Waiting on this until the argTypes design is finalized

File: addons/actions/src/preset/addArgs.ts (hunk 3 above)
Comment (tmeasday) [reply addons/actions/src/preset/addArgs.ts]: I guess not? Just sort of confusing given they are all functions.

**Review by tmeasday — APPROVED** [review]
Lgtm

**Pull request merged**